ManageRDSessions/ManageRDSessions.psm1

```diff
@@ -1,71 +1,108 @@
+            $_.SessionState -like $state -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleTime
+        }
 
-    $sessions = Get-RDUserSession | Where-Object { $_.SessionState -eq 'STATE_DISCONNECTED' -and ( $_.IdleTime / 60000 ) -gt $MinIdleTime }
+    foreach ($session in $sessions) {
+        # Creating PSObject to output and naming as my own type for use in later pipeline input...
+        $object = [PSCustomObject]@{
+            PSTypeName       = "Custom.SB.RDSession"
+            HostServer       = $session.HostServer
+            UserName         = $session.UserName
+            UnifiedSessionID = $session.UnifiedSessionID
+            SessionState     = $session.SessionState
+            IdleTime         = ($session.IdleTime / 60000 -as [int])
+        } #pscustomobject
 
-    foreach ( $session in $sessions ) {
+        # Outputting object
+        Write-Output $object
+    } #foreach
+} #process
 
-        $sessionParams = @{
-            HostServer       = $session.HostServer
-            UnifiedSessionID = $session.UnifiedSessionId
-            Force            = $Force
-        }
-        Invoke-RDUserLogoff @sessionParams
-    }
+End {
+    #Intentionally empty
 }
+} #function
 
-function Disconnect-RDConnectedUserSession {
-    [CmdletBinding()]
+function Remove-sbRDSession {
+    # TO SPLIT INTO 2 DIVERSE FUNCTIONS - trying to do too much (see project)
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
     param (
-        [switch]$Force = $False
+        # Accepting Pipeline ByValue and requiring custom Type - also added [Object[]] to make an array, not sure if needed
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [PSTypeName("Custom.SB.RDSession")][Object[]]$RDSession,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("LogOff", "Disconnect")]
+        [string]$Action
     )
 
-    $sessions = Get-RDUserSession | Where-Object { $_.SessionState -eq 'STATE_ACTIVE' -or $_.SessionState -eq 'STATE_IDLE' }
+    Begin {
+        # Intentionally empty
+    }
+    # Process block is required when receiving objects from the pipeline, otherwise only last object will be received
+    Process {
+        if ($PSCmdlet.ShouldProcess("RDUser: $($RDSession.Username) with Session ID $($RDSession.UnifiedSessionID)", "$Action")) {
 
-    foreach ( $session in $sessions ) {
-        $sessionParams = @{
-            HostServer       = $session.HostServer
-            UnifiedSessionID = $session.UnifiedSessionId
-            Force            = $Force
-        }
-        Disconnect-RDUser @sessionParams
+            $params = @{
+                'HostServer'       = $RDSession.HostServer
+                'UnifiedSessionID' = $RDSession.UnifiedSessionID
+                'ErrorAction'      = 'Stop'
+                'Force'            = $true
+            }
+
+            Write-Verbose "Attempting $Action of $($RDSession.Username) on $($RDSession.HostServer)"
```

Comment:
I'd still surround all vars with brackets here, eg.
`Write-Verbose "Attempting [$Action] of [$($RDSession.Username)] on [$($RDSession.HostServer)]"`

Reply:
Done